### PR TITLE
Add useDisclosure hook

### PR DESCRIPTION
**Pull request by thayannevls**
#### What is the purpose of this pull request?
As the title says.

#### What problem is this solving?
Some components deal with the scenario of open, close, and toggle behavior.

#### How should this be manually tested?
`yarn && yarn styleguide`
#### Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/utilities/index.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { useState } from 'react'
+
+export const useDisclosure = (open = false) => {
+  const [isOpen, setIsOpen] = useState(open)
+
+  return {
+    isOpen,
+    onOpen: () => setIsOpen(true),
+    onClose: () => setIsOpen(false),
+    onToggle: () => setIsOpen(!isOpen),
```
Comment (matheusps) [inline react/utilities/index.ts:10]:
```suggestion
    onToggle: () => setIsOpen(prevIsOpen => !prevIsOpen),
```

Comment (matheusps) [inline react/utilities/index.ts:10]: Wrap these in `useCallback`. This is useful if the users may want to add then to dependency arrays.

File: styleguide.config.js [hunk 2]
```diff
@@ -214,6 +214,15 @@ module.exports = {
         },
       ],
     },
+    {
+      name: 'Utilities',
```
Comment (matheusps) [inline styleguide.config.js:218]: Hooks maybe?

File: react/utilities/index.ts (hunk 1 above)
Comment (thayannevls) [reply react/utilities/index.ts]: Done! :smile:

Comment (thayannevls) [reply react/utilities/index.ts]: Done!

File: styleguide.config.js (hunk 2 above)
Comment (thayannevls) [reply styleguide.config.js:218]: :thinking: Makes sense

@emersonlaurentino do you agree?

File: react/utilities/index.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { useState, useCallback } from 'react'
+
+export const useDisclosure = (open = false) => {
+  const [isOpen, setIsOpen] = useState(open)
+
+  return {
+    isOpen,
+    onOpen: useCallback(() => setIsOpen(true), []),
```
Comment (matheusps) [inline react/utilities/index.ts:8]: Doesn't this usage break the first rule of hooks? 🤔.

Comment (thayannevls) [reply react/utilities/index.ts]: Fixed it

File: styleguide.config.js (hunk 2 above)
Comment (emersonlaurentino) [reply styleguide.config.js:218]: I believe not, because not all utilities are hooks

**Pull request merged**